File: artifacts/original/ProjectOne/src/testing/TaskTest.java
/*
 * Jeremy Cameron
 * CS 320 Milestone Two
 * TaskTest class which is used for conducting tests for the Task class.
 */

package testing;

import static org.junit.jupiter.api.Assertions.*;
import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;

import tasks.Task;

public class TaskTest {
	
	// The Task object used for testing
	private Task task;
	
	// Initializes a new Task object before each test method is run.
	@BeforeEach
	public void setUp() {
		task = new Task("Task Name", "Task Description");
	}
	
	/*
	 * Tests the successful creation of a Task object with valid parameters.
	 * Verifies a unique task ID is generated and that parameters are set correctly.
	 */
	@Test
	public void testTaskCreationSuccess() {
		String id = task.getId();
		assertNotNull(task.getId());
		assertEquals(10, id.length());
		assertEquals("Task Name", task.getName());
		assertEquals("Task Description", task.getDescription());
	}
	
	/*
	 * Tests using an invalid name.
	 * Verifies that an IllegalArgumentException is thrown when the name is longer than 20 characters.
	 */
	@Test
	public void testNameInvalid() {
		assertThrows(IllegalArgumentException.class, () -> {
			new Task("Test Task Name More Than 20 Characters", "Task Description");
		});
	}
	
	/*
	 * Tests using a null value for name.
	 * Verifies that an IllegalArgumentException is thrown when the name is null.
	 */
	@Test
	public void testNameNull() {
		assertThrows(IllegalArgumentException.class, () -> {
			new Task(null, "Task Description");
		});
	}
	
	/*
	 * Tests using an invalid description.
	 * Verifies that an IllegalArgumentException is thrown when the description is longer than 50 characters.
	 */
	@Test
	public void testDescriptionInvalid() {
		assertThrows(IllegalArgumentException.class, () -> {
			new Task("Task Name", "Test Task Description More Than 50 Characters Block Block");
		});
	}
	
	/*
	 * Tests using a null value for description.
	 * Verifies that an IllegalArgumentException is thrown when the description is null.
	 */
	@Test
	public void testDescriptionNull() {
		assertThrows(IllegalArgumentException.class, () -> {
			new Task("Task Name", null);
		});
	}
	
	/*
	 * Tests setting the name at the max character limit (20).
	 * Verifies that the name is updated correctly.
	 */
	@Test
	public void testSetName() {
		task.setName("twentycharacterstest");
		assertEquals("twentycharacterstest", task.getName());
	}
	
	/*
	 * Tests setting the description at the max character limit (50).
	 * Verifies that the description is updated correctly.
	 */
	@Test
	public void testSetDescription() {
		task.setDescription("fiftycharacterstest505050fiftycharacterstest505050");
		assertEquals("fiftycharacterstest505050fiftycharacterstest505050", task.getDescription());
	}
}

File: enhancement_3/CS499_Project/src/Tests/task.test.js
// tests/task.test.js
import Task from '../domain/task.js';

describe('Task', () => {
  let task;

  // Creates a test task before each test
  beforeEach(() => {
    task = new Task({
      name: 'Task Name',
      description: 'Task Description'
    });
  });

  // Test for successfully creating a new task
  test('creates task with valid parameters', () => {
    const id = task.id;
    expect(id).toBeDefined();
    expect(id.length).toBe(10);
    expect(task.name).toBe('Task Name');
    expect(task.description).toBe('Task Description');
  });

  // Test for throwing an error if the task's name is too long
  test('throws error for invalid name (too long)', () => {
    expect(() =>
      new Task({
        name: 'Test Task Name More Than 20 Characters',
        description: 'Task Description'
      })
    ).toThrow();
  });

  // Test for throwing an error if the task's name is null
  test('throws error for null name', () => {
    expect(() =>
      new Task({
        name: null,
        description: 'Task Description'
      })
    ).toThrow();
  });

  // Test for throwing an error if the task's description is too long
  test('throws error for invalid description (too long)', () => {
    expect(() =>
      new Task({
        name: 'Task Name',
        description: 'Test Task Description More Than 50 Characters Block Block'
      })
    ).toThrow();
  });

  // Test for throwing an error if the task's description is null
  test('throws error for null description', () => {
    expect(() =>
      new Task({
        name: 'Task Name',
        description: null
      })
    ).toThrow();
  });

  // Test for successfully updating a task's name and throwing an error if it is too long
  test('setName updates name and throws on invalid input', () => {
    task.setName('twentycharacterstest');
    expect(task.name).toBe('twentycharacterstest');

    expect(() => task.setName(null)).toThrow();
    expect(() => task.setName('This name is way too long for the limit')).toThrow();
  });

  // Test for successfully updating a task's description and throwing an error if it is too long
  test('setDescription updates description and throws on invalid input', () => {
    const validDesc = 'fiftycharacterstest505050fiftycharacterstest505050';
    task.setDescription(validDesc);
    expect(task.description).toBe(validDesc);

    expect(() => task.setDescription(null)).toThrow();
    expect(() =>
      task.setDescription(
        'This description is definitely longer than the fifty characters allowed in the field.'
      )
    ).toThrow();
  });
});
